state-chain/cf-integration-tests/src/trading_strategy.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2025 Chainflip Labs GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+
+use cf_amm::math::price_at_tick;
+use cf_primitives::{
+	AccountId, AccountRole, Asset, AssetAmount, FLIPPERINOS_PER_FLIP, STABLE_ASSET,
+};
+use cf_test_utilities::assert_events_match;
+use frame_support::assert_ok;
+use state_chain_runtime::{AssetBalances, Runtime, RuntimeEvent, RuntimeOrigin};
+
+type TradingStrategyPallet = state_chain_runtime::TradingStrategy;
+use cf_traits::BalanceApi;
+use pallet_cf_trading_strategy::TradingStrategy;
+
+use crate::{
+	network::register_refund_addresses,
+	swapping::{credit_account, do_eth_swap, new_pool},
+};
+
+const DORIS: AccountId = AccountId::new([0x11; 32]);
+const ZION: AccountId = AccountId::new([0x22; 32]);
+const BASE_ASSET: Asset = Asset::Usdt;
+const QUOTE_ASSET: Asset = STABLE_ASSET;
+
+#[test]
+fn basic_usage() {
+	const DECIMALS: u128 = 10u128.pow(6);
+	const AMOUNT: AssetAmount = 10_000 * DECIMALS;
+
+	super::genesis::with_test_defaults()
+		.with_additional_accounts(&[(
+			DORIS,
+			AccountRole::LiquidityProvider,
+			5 * FLIPPERINOS_PER_FLIP,
+		),
+	    (ZION, AccountRole::Broker, 5 * FLIPPERINOS_PER_FLIP),])
+		.build()
+		.execute_with(|| {
+            new_pool(BASE_ASSET, 0, price_at_tick(0).unwrap());
+
+            // Start trading strategy
+            register_refund_addresses(&DORIS);
+            credit_account(&DORIS, BASE_ASSET, AMOUNT);
+            const STRATEGY: TradingStrategy =
+                TradingStrategy::SellAndBuyAtTicks { sell_tick: 1, buy_tick: -1, base_asset: BASE_ASSET };
+            assert_ok!(TradingStrategyPallet::deploy_trading_strategy(
+                RuntimeOrigin::signed(DORIS),
+                STRATEGY.clone(),
+                BTreeMap::from_iter(vec![(BASE_ASSET, AMOUNT)]),
```

Comment:
Nit: you don't have to create a Vec, `from_iter` works on slices too. (Doesn't really matter here in a test, but it is useful to know.)

```suggestion
                BTreeMap::from_iter([(BASE_ASSET, AMOUNT)]),
```

Reply:
Yeah i did realise this later and used just a slice but then forgot to come back and change this one 😅